platform-sdk/docs/proposals/healthMonitor/healthMonitorProposal.md

```diff
@@ -0,0 +1,127 @@
+# Health Monitor Proposal
+
+---
+
+## Summary
+
+A replacement for the current hard backpressure mechanism using a health monitor.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Designers          | Cody Littley                        |
+| Functional Impacts | Consensus node performance          |
+| Related Proposals  | n/a                                 |
+| HIPS               | n/a                                 |
+
+---
+
+## Purpose and Context
+
+The platform's current backpressure mechanism has very bad performance characteristics.
+We need to replace the current backpressure implementation with something more performant.
+
+### Requirements
+
+- The backpressure mechanism must prevent an OOM in the event of any arbitrary component
+  becoming arbitrarily slow.
+- The backpressure mechanism should not slow down critical processes when enganged.
+  Specifically, transaction handling should not be impeded when backpressure is engaged.
+  This includes slowdowns due to the transaction handling thread being starved of work.
+- The backpressure mechanism should not be overly aggressive. That is, we shouldn't see
+  it engaging under normal circumstances on nodes that are not in a state of distress.
+- The implementation of the backpressure mechanism should be sufficiently clean and
+  maintainable. A performant solution that is too complex is not a solution we should
+  accept as a solution.
+
+### Design Decisions
+
+This change can be split into two distinct sub-problems: the detection of a system in duress, and the action that
```

Comment:
I do not understand this chapter? Does that mean that you will create 2 independent apis? 2 proposals? Please be more concreate.